### DISCUSSION TITLE
Separate branch/MRs when touching pyproject.toml (for poetry)

### DISCUSSION THREAD

**Discussion by 4sfaloth** [renovatebot/renovate#28116] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

Hi,

we are using Renovate for some fairly complex projects with lots of dependencies. Recently I have been investing some time intro improving our configurations, since most of the MRs were simply getting stuck. I have managed to improve things a lot already using some simple rules (eg: major updates go on separate branches, linters are grouped separately, patches are auto-merged, etc...) but on the most complex `Python` (`poetry`) projects am still facing some issues due to dependency resolution being broken (due to conflicting constraints). To help untangle this what I would like to do is
1. only group (minor) updates when this can be done adjusting the `.lock` file only (eg: as if `rangeStrategy=in-range-only`)
2. updates that require changing the `pyproject.toml` (eg: only possible with `rangeStrategy=update-lockfile`) should be grouped separately (or even be on a per-package basis)

I was trying to achieve this by using something like
```json
    {
      "matchUpdateTypes": [
        "minor",
      ],
      "groupName": "minor updates (lock only)",
      "rangeStrategy": "in-range-only",
    },
    {
      "matchUpdateTypes": [
        "minor",
      ],
      "groupName": "minor updates",
      "rangeStrategy": "update-lockfile",
    }
```
but it seems like you cannot use `matchUpdateTypes` and `rangeStrategy` in the same rule. It is possible to achieve this in other way?

Also keen in hearing about other alternatives, if my approach does not seem like the best.

Thanks in advance,

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by secustor** [comment, marked as answer]
`rangeStrategy` is not doing what you expect here. It sets the value not matching it. 
Only values which are prefixed with `match` or `exclude` are conditionals all other values set the value. https://docs.renovatebot.com/configuration-options/#packagerules

There is currently no `matchRangeStrategy` option, tough should easy to implement. https://github.com/renovatebot/renovate/pull/27374/files

**Reply by 4sfaloth** [reply]
Thanks for the explanation. I will consider contributing but given I am not familiar with the project I doubt I might find the time 😢
